lib/board.ex

```diff
@@ -12,40 +13,24 @@ defmodule Board do
     |> update_board(board)
   end
 
-  defp update_board({:valid, position}, board) do
-    next_board = board
-               |> next_player_mark
-               |> update(board, position-1)
-    {:ok, next_board}
-  end
-
-  defp update_board(error, _), do: error
-
-  defp update(mark, board, position) do
-    List.replace_at(board, position , mark)
-  end
-
   def next_player_mark(board), do: if more_o_marks(board), do: "X", else: "O"
 
   def mark_count(mark, board), do: Enum.count(board, fn(cell) -> cell == mark end)
 
   def game_over?(board), do: winner?(board) || draw?(board)
 
-  def winner?(board), do: winner?(board, @winning_combinations)
-
-  def draw?(board), do: board_full?(board) && !winner?(board)
-
-  def board_full?(board) do
+  def winner?(board) do
     board
-    |> Enum.reject(&mark?/1)
-    |> Enum.empty?
+    |> current_lines
+    |> Enum.map(fn(line) -> all_same_marks?(line) end)
```

Comment:
This could be shortened to 
`Enum.map(&all_same_marks?/1)`

But it makes me wonder, why don't you use that in the `any` clause like so:
`Enum.any?(&all_same_marks?/1)`? Why map it to true/false and then filter?